### PR TITLE
Updated input files challenge

### PR DESCRIPTION
**Pull request by bhoffman0**
To include student choice using socially useful data files in the ending challenge.
Updates are to the last two sections where I moved the pokemon up one section and added a new challenge that has a choice of different data sets. I'll ask teachers for more files to include. Could include some APIs too. Unit tests are not great.

### CODE REVIEW

File: _sources/Unit7-ArrayList/topic-input-files.rst [hunk 1]
```diff
@@ -550,6 +581,145 @@ In this challenge, you will read in the data from the pokemon file and save it i
...
+   :datafile: pokemon.csv, WorldIndicators2000.csv, StateData2020-CDC-Census.csv, fastfood2023.csv, NBAPlayerStats2021-2022.csv
+   
+   **Input File Challenge:** Design the class for your input file that has at least 3 attributes that can be found in the file. Then, read in the data from the file, split each line, and save the data in an ``ArrayList`` of objects. Finally, do something interesting with the data using a loop, for example you could find the object with the max or min attribute value or print out all the objects of a certain attribute value.
+   ~~~~
+   import java.io.*;
+   import java.util.*;
+   import java.nio.file.*;
+   
+   /* Change YourClassName below */
+   class YourClassName
```
Comment (gigamonkey) [inline _sources/Unit7-ArrayList/topic-input-files.rst:640]:
```suggestion
   /* Change this class name to a meaningful name for one record from the dataset. */
   class OneItem
```

File: _sources/Unit7-ArrayList/topic-input-files.rst [hunk 2]
```diff
@@ -550,6 +581,145 @@ In this challenge, you will read in the data from the pokemon file and save it i
...
+   {
+       // Add at least 3 attributes found in the data file
+         
+       // Add a constructor that initializes the attributes 
+         
+       // Add any getters and toString methods that you need
+
+   }
+
+   public class MyArrayList
```
Comment (gigamonkey) [inline _sources/Unit7-ArrayList/topic-input-files.rst:650]:
```suggestion
   /* Change this class name to a meaningful name for the whole dataset. */
   public class Data 
```

**Review by gigamonkey — APPROVED** [review]
Looks good. A couple suggested changes aimed at guiding students to thinking about their class names a bit more carefully.

File: _sources/Unit7-ArrayList/topic-input-files.rst [hunk 3]
```diff
@@ -550,6 +581,146 @@ In this challenge, you will read in the data from the pokemon file and save it i
...
+
+
+       // Write a method that does something with the data
+       // for example find the object with a min or max attribute value
+       // or print out all the objects of a certain attribute value.
+
+
+       public static void main(String[] args) throws IOException
+       {
+            MyArrayList obj = new MyArrayList();
```
Comment (bhoffman0) [inline _sources/Unit7-ArrayList/topic-input-files.rst:669]:
```suggestion
            Data obj = new Data(); // change to your class name
```

Comment (gigamonkey) [reply _sources/Unit7-ArrayList/topic-input-files.rst]: Oh. Whoops. Yup.

**Pull request merged**